tests/Integration/PrestaShopBundle/Controller/FormFiller/FormChecker.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\PrestaShopBundle\Controller\FormFiller;
+
+use InvalidArgumentException;
+use PHPUnit\Framework\Assert;
+use Symfony\Component\DomCrawler\Field\FormField;
+use Symfony\Component\DomCrawler\Form;
+
+class FormChecker
+{
+    /**
+     * @param Form $form
+     * @param array $expectedFormData
+     *
+     * @return Form
+     */
+    public function checkForm(Form $form, array $expectedFormData): Form
+    {
+        foreach ($expectedFormData as $fieldName => $expectedFormDatum) {
+            if (!is_array($expectedFormDatum)) {
+                /** @var FormField $formField */
+                $formField = $form->get($fieldName);
+                $this->assertFormValue($expectedFormDatum, $formField->getValue(), $fieldName);
+            } else {
+                throw new InvalidArgumentException('The check for array values has not been implemented yet, your turn!!');
```

Comment:
May I suggest
```suggestion
                throw new RuntimeException('Check for array values has not been implemented yet');
```
A little bit more formal 😄

Reply:
Well, in my opinion, the `InvalidArgumentException` is more adapted to the situation, and thus more formal ^^
It indicates the argument you provided is not acceptable and not a suitable input. Of course it's not acceptable right now because the feature was not developed, but it's still invalid right now.

And I didn't implement this one yet because I have no use case and thus can't know what's expected, so instead of doing something badly implemented, I thought leaving a clear error message for the developer which will encounter this use case was the best compromise for now.